src/components/organisms/Badges/Badges.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useMemo } from "react";
+import { User } from "@bugsnag/js";
+
+import { Link } from "react-router-dom";
+import { useFirestoreConnect } from "react-redux-firebase";
+
+import { useConnectRelatedVenues } from "hooks/useConnectRelatedVenues";
+import { useSelector } from "hooks/useSelector";
+
+import { PLAYA_VENUE_NAME } from "settings";
+
+import { CampRoomData } from "types/CampRoomData";
+import { isCampVenue } from "types/CampVenue";
+import { AnyVenue } from "types/Firestore";
+import { Venue } from "types/Venue";
+
+import { WithId } from "utils/id";
+import { orderedVenuesSelector } from "utils/selectors";
+import { venueInsideUrl, venuePreviewUrl } from "utils/url";
+
+export const Badges: React.FC<{
+  user: WithId<User>;
+  currentVenue: WithId<Venue>;
+}> = ({ user, currentVenue }) => {
+  useFirestoreConnect([
+    {
+      collection: "users",
+      doc: user.id,
+      subcollections: [{ collection: "visits" }],
+      storeAs: "userModalVisits",
+    },
+  ]);
+  const visits = useSelector(
+    (state) => state.firestore.ordered.userModalVisits
+  );
```

Comment:
[nit] extract selector functions (even if it is just to a const outside of the component, but still in the same file

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/895/commits/3e72c18e563d7fbd06de2c6c287223f3a54ec7be